components/filter/events.tsx

```diff
@@ -140,87 +150,101 @@ const EventTab = ({ children }: { children: React.ReactNode }) => {
-              );
-            })
-          ) : (
-            <h3>No events found matching the filters</h3>
-          )}
-        </>
+      {isLoading ? (
+        <LoadingIcon />
       ) : (
-        <p className="flex flex-row text-xl">
-          <FaSpinner className="m-icon animate-spin" /> Loading Events...
-        </p>
+        <LoadedEvents events={events} isUpcoming={isUpcoming}></LoadedEvents>
       )}
     </div>
   );
 };
 
+type AllEventsProps = {
+  events: EventTrimmed[];
+  isUpcoming: boolean;
+};
+
+const LoadingIcon: React.FC = () => {
+  return (
+    <p className="flex flex-row text-xl">
+      <FaSpinner className="m-icon animate-spin" /> Loading Events...
+    </p>
+  );
+};
+
+const LoadedEvents: React.FC<AllEventsProps> = ({ events, isUpcoming }) => {
+  return (
+    <>
+      {events.length > 0 ? (
+        events?.map((event, index) => {
+          let eventJsonLd: WithContext<Event> = undefined;
+
+          if (index < EVENTS_JSON_LD_LIMIT && isUpcoming) {
+            eventJsonLd = {
+              "@context": "https://schema.org",
+              "@type": "Event",
+              name: event.title,
+              image: event.thumbnail,
+              startDate: event.startDateTime?.toISOString(),
+              endDate: event.endDateTime?.toISOString(),
+              location: {
+                "@type": "Place",
+                address: {
+                  "@type": "PostalAddress",
+                  addressLocality: CITY_MAP[event.city]?.name,
+                  addressRegion: CITY_MAP[event.city]?.state,
+                  addressCountry: CITY_MAP[event.city]?.country,
+                },
+                name: CITY_MAP[event.city]?.name,
+                url: CITY_MAP[event.city]?.url,
+              },
+              eventStatus: "https://schema.org/EventScheduled",
+              eventAttendanceMode:
+                "https://schema.org/MixedEventAttendanceMode",
+              organizer: sswOrganisation,
+            };
+          }
+          return <Event key={index} jsonLd={eventJsonLd} event={event} />;
+        })
+      ) : (
+        <h3>No events found matching the filters</h3>
+      )}
+    </>
+  );
+};
+
 interface EventProps {
   visible?: boolean;
   event: EventTrimmed;
   jsonLd?: WithContext<Event>;
 }
 
-const Event = ({ visible, event, jsonLd }: EventProps) => {
+const Event = ({ event, jsonLd }: EventProps) => {
   /* TODO: remove this when Tina cloud sync issue is fixed https://github.com/tinacms/tina-cloud/issues/2073
 
   We need this because there's an issue preventing us from syncing the files in the repo
   to Tina cloud. Images that aren't synced will 404.
    
    */
-  const [thumbnail, setFallbackImage] = useState(event.thumbnail);
+  const [oldTitle, setOldTitle] = useState(event.title);
+  const [visible, setVisible] = useState(false);
+  const [thumbnail, setFallbackImage] = useState("");
+  useEffect(() => {
+    if (event.title !== oldTitle) setVisible(false);
```

Comment:
can we put parenthesis around here? as per this rule - https://www.ssw.com.au/rules/how-to-avoid-problems-in-if-statements/#:~:text=See%20history.%20Try%20to

Reply:
Done - I've added parentheses around the if statement